kubespawner/spawner.py

```diff
@@ -2407,7 +2408,7 @@ def on_reflector_failure():
         previous_reflector = self.__class__.reflectors.get(key)
 
         if replace or not previous_reflector:
-            self.__class__.reflectors[key] = ReflectorClass(
+            self.__class__.reflectors[key] = reflector_class(
```

Comment:
Why reflector is stored in the class attribute? Is it caused by some performance issues with Kubernetes API?

Reply:
Ok, added a comment